### PR TITLE
Feature/adding functional to pages

### PR DESCRIPTION
**Pull request by romanazaruk**

### CODE REVIEW

File: frontend/src/components/constants/constants.ts [hunk 1]
```diff
@@ -14,3 +14,11 @@ export const languageLevels = [
...
   'Advanced'
 ];
+
+export enum Sex {
+  Male,
+  Female,
+  Other
+}
+
+export const sex = ['Male', 'Female', 'Other'];
```
Comment (oleksandrstarov) [inline frontend/src/components/constants/constants.ts]: not sure I get this...

File: frontend/src/components/shared/InputWithFormik/InputWithFormik.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { useField } from 'formik';
+import Input from '../input/Input';
+
+type InputProps = {
+  name: string;
+  label: string;
+  type: string;
+  max?: string;
```
Comment (oleksandrstarov) [inline frontend/src/components/shared/InputWithFormik/InputWithFormik.tsx]: what Max stands for? Maybe should be number?

File: frontend/src/components/shared/previewInfo/PreviewInfo.tsx [hunk 3]
```diff
@@ -4,13 +4,59 @@ import Box from '@material-ui/core/Box';
...
+    const birthDate = new Date(dateString);
+    let age = today.getFullYear() - birthDate.getFullYear();
+    const month = today.getMonth() - birthDate.getMonth();
+    if (month < 0 || (month === 0 && today.getDate() < birthDate.getDate())) {
+      age--;
+    }
+    return age;
+  }
+
+  const old = getOld(values.birthday);
```
Comment (oleksandrstarov) [inline frontend/src/components/shared/previewInfo/PreviewInfo.tsx]: Rename to calculateAge
Use moment for this (I believe this is already in the application)

File: frontend/src/components/shared/previewInfo/PreviewInfo.tsx [hunk 4]
```diff
@@ -4,13 +4,59 @@ import Box from '@material-ui/core/Box';
...
+
+  const getMethodPriority = (country: keyof typeof Country) => {
+    return Country[country];
+  };
```
Comment (oleksandrstarov) [inline frontend/src/components/shared/previewInfo/PreviewInfo.tsx:33]: what the purpose for this? Do we ever use this method?

File: frontend/src/components/shared/select/Select.tsx [hunk 5]
```diff
@@ -1,14 +1,29 @@
...
+  onChange?: (e: any) => void;
 };
+
+const getOptions = (arr: (string | number)[]) => {
+  return arr.map((item, i) => (
+    <option key={`${item}${i}`} value={item}>
+      {item}
+    </option>
+  ));
+};
```
Comment (oleksandrstarov) [inline frontend/src/components/shared/select/Select.tsx]: why do we need separate function? It is used only in one place
<select className="select" onChange={props.onChange} name={props.name} value={props.value}>
        {props.options.map((item, i) => (
    <option key={`${item}${i}`} value={item}>
      {item}
    </option>}
      </select>

File: frontend/src/components/shared/userCard/UserCard.tsx [hunk 6]
```diff
@@ -25,21 +25,18 @@ export const UserCard: React.FC<MainInfoProps> = ({ mainInfo }, shadowIntensity)
...
-        <UserPhoto
-          isOnline={mainInfo.isOnline}
-          photoUrl={mainInfo.photoUrl}
-        />
+        <UserPhoto isOnline={mainInfo.isOnline} photoUrl={mainInfo.photoUrl} />
         <h3>
           {mainInfo.firstName} {mainInfo.lastName}
         </h3>
-        <p>{mainInfo.age} y.o.</p>
+        {mainInfo.age ? <p>{mainInfo.age} y.o.</p> : null}
```
Comment (oleksandrstarov) [inline frontend/src/components/shared/userCard/UserCard.tsx]: can be 

{mainInfo.age && <p>{mainInfo.age} y.o.</p>}

File: frontend/src/components/shared/userCard/UserCard.tsx [hunk 7]
```diff
@@ -68,19 +65,18 @@ export const UserCard: React.FC<MainInfoProps> = ({ mainInfo }, shadowIntensity)
...
         </div>
-        {false || loginedUser.id === mainInfo.id ?
-          null
-          : <div className="buttons-action">
-            {mainInfo.isFriend
-              ? null
-              : <Link to="/" onClick={sendFriendRequest} className="add-friend">
+        {false || loginedUser.id === mainInfo.id ? null : (
+          <div className="buttons-action">
+            {mainInfo.isFriend ? null : (
```
Comment (oleksandrstarov) [inline frontend/src/components/shared/userCard/UserCard.tsx]: same as above

File: frontend/src/components/shared/userPhoto/UserPhoto.tsx [hunk 8]
```diff
@@ -3,11 +3,14 @@ import { UserPhotoInfo } from '../../interfaces/Interface';
 import Avatar from '../../mocks/user.png';
 import './userPhoto.scss';
 
-export const UserPhoto: React.FC<UserPhotoInfo> = (photoInfo) => {
+export const UserPhoto: React.FC<UserPhotoInfo> = photoInfo => {
+  function checkURL(url: string) {
+    return url.match(/\.(jpeg|jpg|gif|png)$/) != null || url.startsWith('blob:');
```
Comment (oleksandrstarov) [inline frontend/src/components/shared/userPhoto/UserPhoto.tsx:8]: maybe 
/blob:|\.(jpeg|jpg|gif|png)/.test(url) ?

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

**Review by OrestHaponov — APPROVED** [review]
Lgtm

File: frontend/src/components/shared/previewInfo/PreviewInfo.tsx (hunk 4 above)
Comment (romanazaruk) [reply frontend/src/components/shared/previewInfo/PreviewInfo.tsx:33]: Thats send the country code

File: frontend/src/components/shared/userPhoto/UserPhoto.tsx (hunk 8 above)
Comment (romanazaruk) [reply frontend/src/components/shared/userPhoto/UserPhoto.tsx:8]: That`s a feauture for time when we dont have normal data in backend

**Review by oleksandrstarov — APPROVED** [review]

**Pull request merged**